src/ansys/tools/versioning/utils.py

```diff
@@ -276,3 +283,281 @@ def wrapper(self, *args, **kwargs):
+            else:
+                raise ValueError("Invalid version string")
+        else:
+            return super().__ge__(__x)
+
+    def __gt__(self, __x: Union[str, int]) -> bool:
+        """Greater than.
+
+        If compared against a string which contains 'dev' it will always evaluate to False.
+        If compared against an int, it will perform a classic 'greater than' operation.
+        """
+        if isinstance(__x, str):
+            if "dev" in __x and isinstance(self, str) and "dev" in self:
+                raise ValueError("Two 'dev' versions cannot be compared against.")
+            elif "dev" in __x:
+                return False
+            else:
+                raise ValueError("Invalid version string")
+        else:
+            return super().__gt__(__x)
+
+    def __eq__(self, __x: object) -> bool:
+        """Equal method.
+
+        If compared against a string which contains 'dev' it will always evaluate to False.
+        If compared against an int, it will perform a classic 'equal' operation.
+        """
+        if isinstance(self, str) and isinstance(__x, str) and "dev" in self and "dev" in __x:
+            return str(self) == str(__x)
+
+        elif isinstance(__x, str):
+            return False
+        else:
+            return super().__eq__(__x)
+
+    def __ne__(self, __x: object) -> bool:
+        """Not equal.
+
+        If compared against a string which contains 'dev' it will always evaluate to not
+        'equal' operation (True). If compared against an int, it will perform a classic 'not equal' operation.
+        """
+        if isinstance(__x, str):
+            return not self.__eq__(__x)
+        else:
+            return super().__ne__(__x)
+
+    def __hash__(self) -> int:
+        """Call the underlying __hash__ method."""
+        return super().__hash__()
+
+
+def valid_version_string(version):
+    """Check if version string is valid."""
+    if isinstance(version, str) and (
+        version.lower().replace("dev", "").replace(".", "").isdigit() or version.lower() == "dev"
+    ):
+        return True
+    elif isinstance(version, int):
+        return True
+    else:
+        return False
+
+
+def valid_semantic_version(iterable):
+    """Check if a semantic version is valid."""
+    valid_major_minor = all(
+        isinstance(each, int) or (isinstance(each, str) and each.isdigit()) for each in iterable[:2]
+    )
+    valid_patch = valid_version_string(iterable[2])
+
+    if valid_major_minor and valid_patch:
+        return True
+    else:
+        return False
+
+
+class SemanticVersion(tuple):
+    """
+    Class for semantic versioning.
+
+    It is a subclass of tuple and can be instantiated from a string or a tuple.
+
+    You can use 'dev' in the patch version, but nowhere else.
+
+    Parameters
+    ----------
+    tuple : _type_
+        _description_
+    """
+
+    def __new__(cls: type, __iterable: Iterable = None, major=None, minor=None, patch=None):
```

Comment:
If values are optional, then use the Optional type hint.
```suggestion
    def __new__(cls: type, __iterable: Optional[Iterable] = None, major : Optional[VersionNumber] =None, minor : Optional[VersionNumber] =None, patch : Optional[VersionNumber] =None):
```